Packs/BastilleNetworks/Integrations/BastilleNetworks/BastilleNetworks.py

```diff
@@ -0,0 +1,350 @@
+
+        data = self._build_tag_action_body(site, concentrator, map, transmitter_id, tag)
+
+        return self._http_request(method='POST', url_suffix=url_suffix, json_data=data)
+
+
+def test_module(client):
+    """ Check connection to the Bastille API service """
+
+    try:
+        client.get_zone_events(limit=0)
+        result = 'ok'
+    except DemistoException as exc:
+        if '404' in str(exc):
+            result = '404 Server Not Found'
+        elif '401' in str(exc):
+            result = '401 Unauthorized'
+        elif '500' in str(exc):
+            result = '500 Internal Server Error'
+        else:
+            result = f'Unknown Error {exc}'
+
+    return result
+
+
+def get_site_params():
+    """ Helper to get the area related integration parameters """
+
+    params = demisto.params()
+    site = params.get('site')
+    concentrator = params.get('concentrator')
+    map = params.get('map')
+
+    return site, concentrator, map
+
+
+def get_event_args(args):
+    """ Helper to get the common event related command arguments """
+
+    protocol = args.get('protocol')
+    since = args.get('since')
+    until = args.get('until')
+    limit = args.get('limit')
+    tags = args.get('tags')
+    event_id = args.get('event_id')
+
+    if isinstance(since, str):
+        since = dateparser.parse(since, settings={'TIMEZONE': 'UTC'}).timestamp()
+    if isinstance(until, str):
+        until = dateparser.parse(until, settings={'TIMEZONE': 'UTC'}).timestamp()
+
+    return protocol, since, until, limit, tags, event_id
+
+
+def format_timestamp(timestamp):
+    """ Convert unix epoch to ISO formatted timestamp """
+
+    return datetime.utcfromtimestamp(timestamp).astimezone().isoformat()
+
+
+def parse_events(events, readable_fields):
+    """ Helper to parse common event command response """
+
+    for event in events:
+        event['time_s'] = format_timestamp(event['time_s'])
+        event['first_seen']['time_s'] = format_timestamp(event['first_seen']['time_s'])
+        if 'last_seen' in event:
+            event['last_seen']['time_s'] = format_timestamp(event['last_seen']['time_s'])
+
+    readable_events = [{f: e[f] for f in readable_fields} for e in events]
+
+    return events, readable_events
+
+
+def get_zone_events_command(client, args):
+    """ Command to fetch zone events """
+
+    site, concentrator, map = get_site_params()
+    protocol, since, until, limit, tags, event_id = get_event_args(args)
+    zone = args.get('zone')
+
+    events = client.get_zone_events(site=site, concentrator=concentrator, map=map,
+                                    zone=zone, protocol=protocol, since=since,
+                                    until=until, limit=limit, tags=tags,
+                                    event_id=event_id)
+
+    readable_fields = ['event_id', 'time_s', 'area', 'zone_name', 'tags', 'device_info',
+                       'emitter', 'first_seen', 'last_seen']
+
+    raw_response = json.dumps(events)
```

Comment:
why do you need the json.dumps here? we need the real json
Here and everywhere else